src/front/spv/image.rs

```diff
@@ -432,26 +441,42 @@ impl<I: Iterator<Item = u32>> super::Parser<I> {
         let coord_lexp = self.lookup_expression.lookup(coordinate_id)?;
         let coord_type_handle = self.lookup_type.lookup(coord_lexp.type_id)?.handle;
 
-        let image_var_handle = expressions.get_global_var(si_lexp.image)?;
-        let sampler_var_handle = expressions.get_global_var(si_lexp.sampler)?;
-        log::debug!(
-            "\t\t\tImage {:?} sampled with {:?} under {:?}",
-            image_var_handle,
-            sampler_var_handle,
-            options,
-        );
         let sampling_bit = if options.compare {
             SamplingFlags::COMPARISON
         } else {
             SamplingFlags::REGULAR
         };
-        if let Some(flags) = self.handle_sampling.get_mut(&image_var_handle) {
-            *flags |= sampling_bit;
+
+        let image_ty = match expressions[si_lexp.image] {
+            crate::Expression::GlobalVariable(handle) => {
+                if let Some(flags) = self.handle_sampling.get_mut(&handle) {
+                    *flags |= sampling_bit;
+                }
+
+                global_arena[handle].ty
+            }
+            crate::Expression::FunctionArgument(i) => {
+                let flags = function_info.parameters_sampling[i as usize]
```

Comment:
isn't the parameters sampling required to have the size equal to the number of arguments already?

Reply:
Yes, I think you are confused about the `get_or_insert` right? The type of `parameters_sampling` is `Vec<Option<image::SamplingFlags>>` so the `get_or_insert` refers to the `Option` so we are getting a mutable reference whilst making it `Some` if it wasn't already.